feat: add CRaC priming support to powertools-kafka module

kjswaruph opened the pull request:
## Summary

Add Crac support for kafka module of powertools.

### Changes

> Please provide a summary of what's being changed

<!-- What is this PR solving? Write a clear description or reference the issue(s) it addresses. -->

- Add powertools-common dependency, maven test profile and classesloaded.txt for preloading
- Add Crac dependency and update PowertoolsSerializer to register as Crac Resource
- Add tests in PowertoolsSerializerTest to assert beforeCheckpoint and afterRestore hooks do not throw exception

> Please add the issue number below, if no issue is present the PR might get blocked and not be reviewed

**Issue number:** #1999 

<!-------
Before creating the pull request, please make sure you do the following:

- Read the Contributing Guidelines at https://github.com/aws-powertools/powertools-lambda-java/blob/main/CONTRIBUTING.md#sending-a-pull-request
- Check that there isn't already a PR that addresses the same issue. If you find a duplicate, please leave a comment under the existing PR so we can discuss how to move forward
- Check that the change meets the project's tenets https://docs.powertools.aws.dev/lambda/java/latest/#tenets
- Add a PR title that follows the conventional commit semantics - https://www.conventionalcommits.org/en/v1.0.0/
- If relevant, add tests that prove that the change is effective and works
- Whenever relevant, make sure to comment functions/methods/types and make appropriate changes to the documentation
------->

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

svozza commented:
Hi @kjswaruph. Thank you for the PR! I have assigned the PR to you and we will begin reviewing it and giving feedback soon.

kjswaruph commented:
Hi @phipag, sorry for responding late. I’ll update the PR with the Avro GenericRecord as recommended and aim to finish it by this weekend. If I encounter any issues or need clarification, I’ll reach out. Thanks again for your feedback!

kjswaruph commented:
@phipag Sorry for the earlier confusion with the pr, I accidentally messed up the history while resolving merge conflicts. I have now cleaned up the branch and removed the Avro priming as requested. Please review again and let me know if there’s anything else needed.

phipag commented on powertools-kafka/src/main/java/software/amazon/lambda/powertools/kafka/PowertoolsSerializer.java:
```diff
@@ -43,6 +51,13 @@ DeserializationType.KAFKA_PROTOBUF, new KafkaProtobufDeserializer(),
+    private static final PowertoolsSerializer INSTANCE = new PowertoolsSerializer();
+
+    // CRaC registration happens at class loading time
+    static{
```
```suggestion
    static {
```

phipag reviewed (changes requested):
Thanks @kjswaruph. This looks good to me. Can you please commit these two formatting fixes?
